clarifai/models/model_serving/models/default_test.py

```diff
@@ -0,0 +1,230 @@
+    self.triton_model_config = self.triton_model.config_msg
+    self.triton_model_input_name = self.triton_model.input_name
+    self.preprocess = self._get_preprocess()
+    # load labels
+    self._required_label_model_types = [
+        "visual-detector", "visual-classifier", "text-classifier", "visual-segmenter"
+    ]
+    self.labels = []
+    if self.model_type in self._required_label_model_types:
+      with open(os.path.join(repo_version_dir, "../labels.txt"), 'r') as fp:
+        labels = fp.readlines()
+        if labels:
+          self.labels = [line for line in labels if line]
+
+  def test_triton_config(self):
+    """ test Triton config"""
+    # check if input names are still matched
+    self.assertEqual(
+        self.triton_model_input_name, self.default_triton_model_config.input[0].name,
+        "input name of current model vs generated model must be matched "
+        f"{self.triton_model_input_name} != {self.default_triton_model_config.input[0].name}")
+    # check if output names are still matched
+    default_output_names = [each.name for each in self.default_triton_model_config.output]
+    for output_name in self.triton_model_config.output:
+      self.assertIn(output_name.name, default_output_names,
+                    "output name of current model vs generated model must be matched "
+                    f"{output_name.name} not in {default_output_names}")
+
+  def test_having_labels(self):
+    if self.model_type in self._required_label_model_types:
+      self.assertTrue(
+          len(self.labels),
+          f"`labels.txt` is empty!. Model type `{self.model_type}` requires input labels in `labels.txt`"
+      )
+
+  def test_inference_with_predefined_inputs(self):
+    """ Test Inference with predefined inputs """
+
+    if self.preprocess:
+      inputs = [self.preprocess(inp) for inp in PREDEFINED_IMAGES]
+    elif "image" in self.triton_model_input_name:
+      inputs = PREDEFINED_IMAGES
+      logging.info(inputs[0].shape)
+    else:
+      inputs = PREDEFINED_TEXTS
+    outputs = [self.triton_get_predictions(inp) for inp in inputs]
+
+    # Test for specific model type:
+    # 1. length of output array vs config
+    # 2. type of outputs
+    # 3. test range value, shape and dtype of output
+    def _is_valid_logit(x: np.array):
+      return np.all(0 <= x) and np.all(x <= 1)
+
+    def _is_non_negative(x: np.array):
+      return np.all(x >= 0)
+
+    def _is_integer(x):
+      return np.all(np.equal(np.mod(x, 1), 0))
+
+    for inp, output in zip(inputs, outputs):
+
+      field = dataclasses.fields(output)[0].name
+      self.assertEqual(
+          len(self.triton_model_config.output[0].dims),
+          len(getattr(output, field).shape),
+          "Length of 'dims' of config and output must be matched, but get "
+          f"Config {len(self.triton_model_config.output[0].dims)} != Output {len(getattr(output, field).shape)}"
+      )
+
+      if self.model_type == "visual-detector":
```

Comment:
One possibility for these cases is to call predict and check outputs separately. E.g.: visual-classifier and text-classifier can use the same output assertions. text-classifier and text-to-text can use the same input formatting.

Reply:
Just a suggestion, not necessary to do now

Reply:
Thank you for your suggestion.